build-farm/platform-specific-configurations/linux.sh

```diff
@@ -207,7 +207,7 @@ if [ "${VARIANT}" == "${BUILD_VARIANT_DRAGONWELL}" ] && [ "$JAVA_FEATURE_VERSION
   export CXX=/usr/local/gcc9/bin/g++-9.3
   # Enable GCC 10 for Java 17+ for repeatable builds, but not for our supported releases
   # Ref https://github.com/adoptium/temurin-build/issues/2787
-elif [ "$JAVA_FEATURE_VERSION" -ge 19 ] && [ -r /usr/local/gcc11/bin/gcc-11.2 ] && [ "${VARIANT}" != "${BUILD_VARIANT_OPENJ9}" ] ; then
+elif [ "$JAVA_FEATURE_VERSION" -ge 19 ] && [ -r /usr/local/gcc11/bin/gcc-11.2 ] && [ "${VARIANT}" != "${BUILD_VARIANT_OPENJ9}" ] || [ "${ARCHITECTURE}" != "aarch64" ] ; then
```

Comment:
Should this be `== "${BUILD_VARIANT_OPENJ9}"` ?

Reply:
Adopt may be compiling OpenJ9 on 11.2 all platforms but I can't see the jobs
https://github.com/adoptium/temurin-build/blob/master/build-farm/platform-specific-configurations/linux.sh#L210
```
 # Enable GCC 10 for Java 17+ for repeatable builds, but not for our supported releases
  # Ref https://github.com/adoptium/temurin-build/issues/2787
elif [ "$JAVA_FEATURE_VERSION" -ge 19 ] && [ -r /usr/local/gcc11/bin/gcc-11.2 ]; then
  export PATH=/usr/local/gcc11/bin:$PATH
  [ -r /usr/local/gcc11/bin/gcc-11.2 ] && export  CC=/usr/local/gcc11/bin/gcc-11.2
  [ -r /usr/local/gcc11/bin/g++-11.2 ] && export CXX=/usr/local/gcc11/bin/g++-11.2
  export LD_LIBRARY_PATH=/usr/local/gcc11/lib64:/usr/local/gcc11/lib
```

I guess the `if` should be `ge 19 && ! OpenJ9 on alinux`

Reply:
I'm not sure if you still want a change. The code is ! OpenJ9 OR ! aarch64, which should cover x, p, z.

If Adopt is compiling on alinux with 11.2 they must be using `--disable-ddr` or it doesn't work.

Reply:
Ok. Sneaky sneaky. Took me a minute to wrap my head around that. I'd suggest we either add a comment or reverse it so it reads easier for the rest of us 😃.

Reply:
Reverse it how? ! aarch64 OR ! OpenJ9 ?

Reply:
Reversed those conditions and added a comment.

Reply:
I meant `ge 19` && `gcc11` && ! (OpenJ9 && alinux)`. But I'm not sure how that syntax would be, if even possible with shell.
I think it's probably fine with the comment.